src/util/catchHandler.ts

```diff
@@ -1,6 +1,16 @@
 import { Debugger } from 'debug';
+import { HttpErrors } from './constants';
+
+interface HttpError extends Error {
+  status: number;
+}
+
+export const catchHandler = (debug: Debugger) => (error: HttpError): Promise<unknown> => {
+  if (error.status < HttpErrors.SERVER_ERROR && error.status !== HttpErrors.RESOURCE_NOT_ACCESSIBLE) {
```

Comment:
A little uncomfortable that we're allowing any 4xx through as a non-fatal error. This includes 401 Unauthorized, 404 not founds, etc.

Reply:
true. want me to consider only 403 ?

Reply:
404 could be a reviewer not found, not sure we wanna fail on that. it is a possible outcome that is not that important.

Reply:
I think we should do this on an allow-list basis. Right now we know that 422's are non-fatal, and in the future, if we encounter more stats codes that should be deemed non-fatal, we can add to that allowlist.

I don't think allowing all errors through and selectively picking errors as fatal is the right move here

Reply:
Reason being, it'll make it harder to catch false negatives (where no error was thrown but it should've been fatal)

Reply:
sounds logical. I will do that